core/src/main/java/org/bitcoinj/wallet/DefaultRiskAnalysis.java

```diff
@@ -136,8 +136,11 @@ public enum RuleViolation {
     public static RuleViolation isStandard(Transaction tx) {
         // TODO: Finish this function off.
         if (tx.getVersion() > Transaction.MAX_STANDARD_VERSION || tx.getVersion() < Transaction.MIN_STANDARD_VERSION) {
-            log.warn("TX considered non-standard due to unknown version number {}", tx.getVersion());
-            return RuleViolation.VERSION;
+            if (tx.getVersionShort() > Transaction.MAX_STANDARD_VERSION || tx.getVersionShort() < Transaction.MIN_STANDARD_VERSION ||
+                tx.getType().getValue() >= Transaction.Type.TRANSACTION_TYPE_MAX.getValue()) {
+                log.warn("TX considered non-standard due to unknown version number {} or type {}", tx.getVersionShort(), tx.getType());
+                return RuleViolation.VERSION;
+            }
```

Comment:
without this change, a version 3 transaction with a type other than 0 would be counted as non-standard.   One such example would be a coinbase transaction (type = 5).